pkg/operator/render.go

```diff
@@ -351,3 +352,7 @@ func onPremPlatformAPIServerInternalIPs(cfg mcfgv1.ControllerConfigSpec) (interf
 	}
 	return nil, fmt.Errorf("")
 }
+
+func isSingleReplicaTopology(cfg mcfgv1.ControllerConfigSpec) bool {
+	return cfg.Infra.Status.InfrastructureTopology == configv1.SingleReplicaTopologyMode
```

Comment:
note that there is also `ControlPlaneTopology`, I've chosen `InfrastructureTopology` but I don't have a strong opinion.